toolkit/tools/imagegen/configuration/parse_partition.go

```diff
@@ -0,0 +1,249 @@
+)
+
+var (
+	diskInfo                map[string]int
+	curDiskIndex            int
+	latestDiskIndex         int
+	disks                   []Disk
+	partitionSettings       []PartitionSetting
+	partCmdProcess          map[string]func(string) error
+	newDiskPartition        *Partition
+	newDiskPartitionSetting *PartitionSetting
+)
+
+func initializePrerequisitesForParser() {
+	// Create a mapping between the disk path and the index of the disk in the
+	// Disk array so that when we parse through the partition commands, we can
+	// determine which disk the parition instruction is targeted
+	diskInfo = make(map[string]int)
+
+	// Intialize a placeholder disk for the parser to populate the first partition
+	disks = []Disk{Disk{}}
+	partitionSettings = []PartitionSetting{}
+	curDiskIndex = 0
+	latestDiskIndex = 0
+
+	// Create a mapping of partition flags and corresponding processing functions
+	populatepartCmdProcessMap()
+}
+
+func populatepartCmdProcessMap() {
+	partCmdProcess = make(map[string]func(string) error)
+
+	// Currently only process "ondisk", "ondrive", "size" and "fstype"
+	// Expand this map if supporting more commands in the future
+	partCmdProcess[kickstartPartitionOnDisk] = processDisk
+	partCmdProcess[kickstartPartitionOnDrive] = processDisk
+	partCmdProcess[kickstartPartitionSize] = processPartitionSize
+	partCmdProcess[kickstartPartitionFsType] = processPartitionFsType
+}
+
+func processDisk(InputDiskValue string) (err error) {
+	diskValue := strings.TrimSpace(InputDiskValue)
+	if diskValue == "" {
+		return fmt.Errorf("--ondisk/--ondrive must not be empty")
+	}
+
+	// Check whether this disk has already been parsed or not
+	curIdx, ok := diskInfo[diskValue]
+	if ok {
+		curDiskIndex = curIdx
+	} else {
+		// Don't create new disk if processing on the initial placeholder disk
+		if len(disks) != 1 || len(diskInfo) != 0 {
+			disks = append(disks, Disk{})
+		}
+
+		disks[latestDiskIndex].PartitionTableType = PartitionTableTypeGpt
+
+		// Set TargetDisk and TargetDiskType for unattended installation
+		disks[latestDiskIndex].TargetDisk.Type = "path"
+		disks[latestDiskIndex].TargetDisk.Value = diskValue
+
+		diskInfo[diskValue] = latestDiskIndex
+		curDiskIndex = latestDiskIndex
+		latestDiskIndex++
+	}
+
+	return
+}
+
+func processPartitionSize(InputPartSize string) (err error) {
+	partSize := strings.TrimSpace(InputPartSize)
+
+	if len(disks[curDiskIndex].Partitions) == 0 {
+		newDiskPartition.Start = 1
+	} else {
+		newDiskPartition.Start = disks[curDiskIndex].Partitions[len(disks[curDiskIndex].Partitions)-1].End
+	}
+
+	partitionSize, err := strconv.ParseUint(partSize, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	if !newDiskPartition.Grow {
+		newDiskPartition.End = newDiskPartition.Start + partitionSize
+	}
+
+	return
+}
+
+func processPartitionFsType(inputFsType string) (err error) {
+	fstype := strings.TrimSpace(inputFsType)
+	if fstype == "" {
+		return fmt.Errorf("--fstype must not be empty")
+	}
+
+	if fstype == "" {
```

Comment:
Are we able to check if fstype is one of the supported values? If yes, it would be a good idea imho.

Reply:
I'm doing the check here since this is the parser which only cares about translating the kickstart syntax to json structures correctly. The validity of the input will be checked by the actual partitioning code later on.